presto-orc/src/main/java/com/facebook/presto/orc/stream/ByteOutputStream.java

```diff
@@ -107,18 +108,17 @@ public void writeByte(byte value)
 
     private void flushSequence()
     {
-        if (size == 0) {
-            return;
-        }
-
         if (runCount >= MIN_REPEAT_SIZE) {
             buffer.writeByte(runCount - MIN_REPEAT_SIZE);
             buffer.writeByte(lastValue);
         }
         else {
             buffer.writeByte(-size);
-            for (int i = 0; i < size; i++) {
-                buffer.writeByte(sequenceBuffer[i]);
+            if (size == 1) {
+                buffer.writeByte(sequenceBuffer[0]);
+            }
+            else {
+                buffer.writeBytes(sequenceBuffer, 0, size);
```

Comment:
This is the main change, writing the array instead of writing bytes one-by-one in a loop.